backend/ws_server/ws_server.py

```diff
@@ -59,73 +65,79 @@ def start_server(address, port):
     try:
         loop.run_forever()
     except KeyboardInterrupt:
+        info('cleaning up.', INFO_ID)
         pass
     finally:
         server.close()
         loop.close()
 
 
-class MyServerProtocol(WebSocketServerProtocol):
+class GameServerProtocol(WebSocketServerProtocol):
 
     def onConnect(self, request):
         # onConnect happens before onOpen().  It isn't as useful
         # because onConnect() happens before the connection has succeeded.
         # if you want to do something when a client connects, you probably
         # want to do it in onOpen().  This is more like "onAttempt()".
-        print("Client connecting: {0}".format(request.peer))
+        info("Client connecting: {0}".format(request.peer), INFO_ID)
 
     def onOpen(self):
         connected.append(self)
-        if gameloop_client is not None:
-            gameloop_client.sendMessage(utf("player connected!"), False)
-        print("connections: " + str(len(connected)))
-        self.sendMessage(utf("welcome to the club!"), False)
-
-        if len(connected) == 1:
-            self.sendMessage(utf("you're the first one here! welcome!"), False)
-        elif len(connected) > 1:
-            connected[0].sendMessage(
-                utf("hey, first guy! someone else connected!"), False)
+        if gameloop_client:
+            gameloop_client.sendMessage(format_msg(
+                'a user has connected', MSG.info), False)
+        info("connections: " + str(len(connected)), INFO_ID)
+        self.sendMessage(format_msg(
+            'you have joined the server.', MSG.info), False)
 
     def onMessage(self, payload, isBinary):
-        # for now, only send text messages, no binary yet
         assert isBinary is False
-        if isBinary:
-            # keep this check in for future reference
-            pass
+        as_string = payload.decode('utf8')
+        message = obj_from_json(as_string)
+        m_type = message['type']
```

Comment:
I get this stack trace when running
> python start_test_game_server.py

and connecting from web page:

>Exception in callback WebSocketAdapterProtocol._consume.<locals>.process(<Future finished result=None>) at /home/z/Documents/projects/myRepos/Towers/virtenv-towers/
lib/python3.4/site-packages/autobahn/asyncio/websocket.py:105
handle: <Handle WebSocketAdapterProtocol._consume.<locals>.process(<Future finished result=None>) at /home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/py
thon3.4/site-packages/autobahn/asyncio/websocket.py:105>
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/events.py", line 120, in _run
    self._callback(*self._args)
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/asyncio/websocket.py", line 109, in process
    self._dataReceived(data)
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/websocket/protocol.py", line 1179, in _dataReceived
    self.consumeData()
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/websocket/protocol.py", line 1191, in consumeData
    while self.processData() and self.state != WebSocketProtocol.STATE_CLOSED:
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/websocket/protocol.py", line 1558, in processData
    fr = self.onFrameEnd()
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/websocket/protocol.py", line 1679, in onFrameEnd
    self._onMessageEnd()
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/asyncio/websocket.py", line 154, in _onMessageEnd
    res = self.onMessageEnd()
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/websocket/protocol.py", line 638, in onMessageEnd
    self._onMessage(payload, self.message_is_binary)
  File "/home/z/Documents/projects/myRepos/Towers/virtenv-towers/lib/python3.4/site-packages/autobahn/asyncio/websocket.py", line 159, in _onMessage
    res = self.onMessage(payload, isBinary)
  File "/home/z/Documents/projects/myRepos/Towers/backend/ws_server/ws_server.py", line 98, in onMessage
    m_type = message['type']
TypeError: 'bool' object is not subscriptable